### PR TITLE
[JENKINS-27631] Stronger tests

### PR DESCRIPTION
**Pull request by jglick**
[JENKINS-27631](https://issues.jenkins-ci.org/browse/JENKINS-27631)

I lay awake in bed wondering whether the contextual `EnvVars` got stored somewhere unsafe. Turns out it does, though only temporarily. This PR just makes the test coverage more interesting.

@reviewbybees

### CODE REVIEW

File: src/test/java/org/jenkinsci/plugins/credentialsbinding/impl/BindingStepTest.java [hunk 1]
```diff
@@ -148,24 +171,46 @@
...
+        for (File kid : kids) {
+            String qualifiedName = prefix + kid.getName();
+            if (kid.isDirectory()) {
+                grep(kid, text, qualifiedName + "/", matches);
+            } else if (kid.isFile() && FileUtils.readFileToString(kid).contains(text)) {
+                matches.add(qualifiedName);
+            }
+        }
+    }
+
```
Comment (tfennelly) [inline src/test/java/org/jenkinsci/plugins/credentialsbinding/impl/BindingStepTest.java]: Seems like the above two `grep` methods could be useful utils to make available from the core test-harness.

**Comment by tfennelly** [comment]
:+1:

File: src/test/java/org/jenkinsci/plugins/credentialsbinding/impl/BindingStepTest.java (hunk 1 above)
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/credentialsbinding/impl/BindingStepTest.java]: Perhaps. I cannot recall needing this kind of thing before so I just wrote it as a one-off.

(BTW just the first overload would be `public`—the second is a helper method for the recursion. No easy way in Java to declare a local method, alas.)

File: src/test/java/org/jenkinsci/plugins/credentialsbinding/impl/BindingStepTest.java [hunk 2]
```diff
@@ -77,24 +78,45 @@ public class BindingStepTest {
...
+                SemaphoreStep.waitForStart("basics/1", b);
+            }
+        });
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                WorkflowJob p = story.j.jenkins.getItemByFullName("p", WorkflowJob.class);
+                assertNotNull(p);
+                WorkflowRun b = p.getBuildByNumber(1);
+                assertNotNull(b);
+                assertEquals("TODO JENKINS-27631", Collections.singleton("program.dat"), grep(b.getRootDir(), password));
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/credentialsbinding/impl/BindingStepTest.java:109]: ⇒ #6

**Pull request merged**